### PR TITLE
Enabling Suspended->WaitForApproval transition

### PR DESCRIPTION
**Pull request by TonioGela**

### CODE REVIEW

**Review by beetlecrunch — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/purposemanagement/model/persistence/Adapters.scala [hunk 1]
```diff
@@ -95,7 +95,7 @@ object Adapters {
 
     val ACTIVABLE_STATES             = Seq(Draft, Suspended, WaitingForApproval)
     val SUSPENDABLE_STATES           = Seq(Active, Suspended)
-    val WAITABLE_FOR_APPROVAL_STATES = Seq(Draft)
+    val WAITABLE_FOR_APPROVAL_STATES = Seq(Draft, Suspended)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/purposemanagement/model/persistence/Adapters.scala:98]: Not sure if I'm missing a step here, but a `Suspended` version cannot change to `WaitingForApproval`.

The flow should be
1. Activate version V1
2. Suspend version V1
3. Create a Draft version V2
4. Set V2 to WaitingForApproval
5. Enable version V2 (to Suspended in this case) and Archive V1

**Review by galales — CHANGES_REQUESTED** [review]

File: src/main/scala/it/pagopa/interop/purposemanagement/model/persistence/Adapters.scala (hunk 1 above)
Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/purposemanagement/model/persistence/Adapters.scala:98]: Hmm, I'm not sure I follow you. 

The actions to reproduce the bug are the following:
- Create V1 with a over quota dailyCalls value (V1 in WAITING_FOR_APPROVAL)
- Wait for the producer to accept it (V1 in ACTIVE)
- Increase the dailyCalls quota (V2 in WAITING_FOR_APPROVAL)
- Suspend V2 (V2 in SUSPENDED)
- Re-Activating V2 (and here the V2 SUSPENDED -> V2 WAITING_FOR_APPROVAL transition fails)

If I'm not wrong to support this use case we need to add `Suspended` in the allowed starting point for a `_ => WAITING_FOR_APPROVAL` transition, right?

Comment (galales) [reply src/main/scala/it/pagopa/interop/purposemanagement/model/persistence/Adapters.scala:98]: Not quite, only a Draft version can be changed to WaitingForApproval.

From the steps you pointed I see an unexpected case: the V2 in WAITING_FOR_APPROVAL cannot pass to SUSPENDED.
Only an Active or Suspended version can be changed to Suspended.

A WaitingForApproval version can be only Activated (approved by Producer) or deleted (by Consumer)

Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/purposemanagement/model/persistence/Adapters.scala:98]: hmmm, this requires further investigation so

**Pull request closed without merging**